### ISSUE TITLE
Weekly Scheduled items not firing after current day

### ISSUE THREAD

**Issue by haydockjp** [node-schedule/node-schedule#293]
I am not sure if this an issue with the code, or my lack of understanding, but this is what I have experienced. 

I created a basic script to see if node-schedule could do what I need to do.

NOTE: I have removed some of the code and logging from the attached files, as I do not want to expose the web services that I am calling

This is my schedule definition
[schedule.json.txt](https://github.com/node-schedule/node-schedule/files/416696/schedule.json.txt)

The main script, parses the json file and and adds weekly scheduled items to turn something on and off, via a web service. At the end of the file, it loops through the jobs that have been created and outputs the value of the jobs nextInvocation().

[app.js.txt](https://github.com/node-schedule/node-schedule/files/416698/app.js.txt)

What I have found repeatedly, is the schedule does not continue to fire. I am not sure if the issue is
- Only fires on the day that it was scheduled
- Time Zone / GMT issue
- Only fires the first couple of jobs

Here is the output log
[app.log.txt](https://github.com/node-schedule/node-schedule/files/416699/app.log.txt)

As you can see, it fired the first two, but the next day, nothing happened.

If I am using node-schedule incorrectly, any pointer would be greatly appreciated!

**Comment by haydockjp** [comment]
Also, this is running in a RPi 2B, node.js 4.2.2

**Comment by santigimeno** [comment]
@haydockjp thanks for the report. Would it be possible that you could reduce your issue to a simple test case? It would help debugging the issue a lot. Anyway, I'll try to look at it as time permits.

**Comment by haydockjp** [comment]
Is this simper enough?

Get current date
Loop 10 times
Add four hours each look, and set a scheduled job to recur based on the day of the week, the hour and the minute.

It doesn't seem to fire at all

[testapp.js.txt](https://github.com/node-schedule/node-schedule/files/424670/testapp.js.txt)
[testapp.log.txt](https://github.com/node-schedule/node-schedule/files/424671/testapp.log.txt)


Date Started: Wed Aug 17 2016 13:25:30 GMT-0700 (PDT)
Date Checked: Thu 18 Aug 03:34:48 PDT 2016


---
RPi 2B, Jessie

uname -a
Linux home-pi 4.1.17-v7+ #838 SMP Tue Feb 9 13:15:09 GMT 2016 armv7l GNU/Linux

node -v
v4.2.2

node-schedule: 1.1.1

**Comment by santigimeno** [comment]
@haydockjp run your test in Linux and OS X, and it works correctly. Can you check in a different box than your rpi?

**Comment by RalfZhang** [comment]
Some issue happened on my VPS. ( CentOS 6 )

I set a schedule job with '0 * * * *', which mean it would do something every hour.
But after I detached screen session, this job just run one time. 
When I reattached the screen session, the jobs, which should run before but didn't run, run immediately.

For test, I set another schedule job with '*/10 * * * *', which means it would do something every 10 mins.
It run very well.

Very interesting...

I will do more test. If it succeeds, I think I can offer more info.
